pkg/controller/nodes/task/resourcemanager/config/config.go

```diff
@@ -38,9 +41,53 @@ type RedisConfig struct {
 	HostPaths   []string `json:"hostPaths" pflag:",Redis hosts locations."`
 	PrimaryName string   `json:"primaryName" pflag:",Redis primary name, fill in only if you are connecting to a redis sentinel cluster."`
 	// deprecated: Please use HostPaths instead
-	HostPath   string `json:"hostPath" pflag:",Redis host location"`
-	HostKey    string `json:"hostKey" pflag:",Key for local Redis access"`
-	MaxRetries int    `json:"maxRetries" pflag:",See Redis client options for more info"`
+	HostPath   string    `json:"hostPath" pflag:",Redis host location"`
+	HostKey    string    `json:"hostKey" pflag:",Key for local Redis access"`
+	MaxRetries int       `json:"maxRetries" pflag:",See Redis client options for more info"`
+	TLSConfig  TLSConfig `json:"tlsConfig" pflag:",See the crytpo/tls config object for more info"`
+}
+
+// Specific configs for Redis TLS
+// Ref: https://pkg.go.dev/crypto/tls#Config for informations on how to fill in these fields.
+type TLSConfig struct {
+	ServerName string `json:"serverName" pflag:",Used to verify the hostname."`
+	MinVersion string `json:"minVersion" pflag:",Minimum TLS version that is acceptable"`
+	MaxVersion string `json:"maxVersion" pflag:",Maximum TLS version that is acceptable."`
+}
+
+func GetTLSVersion(version string) uint16 {
+	// Parses string version specifiers into correct tls Version specifier.
+	// Returns zero if unsuccessful which tls.Config will interpret as respective
+	// default for minimum or maximum version.
+	switch version {
+	case "1.0":
+		return tls.VersionTLS10
+	case "1.1":
+		return tls.VersionTLS11
+	case "1.2":
+		return tls.VersionTLS12
+	case "1.3":
+		return tls.VersionTLS13
+	}
+	fmt.Printf("Received unsupported TLS Version %s, reverting to default TLS settings.", version)
+	return 0
+}
+
+func ParseTLSConfig(cfg TLSConfig) *tls.Config {
+	// Parses TLSConfig settings into a proper tls.Config object. Returns a pointer
+	// to nil if no settings were manually defined (i.e. cfg is empty) to keep tls disabled.
+	if cfg != (TLSConfig{}) {
+		/* #nosec G402 */
+		return &tls.Config{
+			//Certificates []Certificate,
+			//RootCAs *x509.CertPool,
+			ServerName: cfg.ServerName,
```

Comment:
Is it enough to configure server name and not CA & Certs? is this just to work around common name mismatch issues?